pkg/controller/infrastructure/infraflow/ensurer.go

```diff
@@ -632,9 +634,10 @@ func (fctx *FlowContext) ensureSubnets(ctx context.Context) (err error) {
 		return *s.Name
 	})
 	// clean the current inventory and rebuild it.
-	for _, name := range fctx.inventory.ByKind(KindSubnet) {
-		if subnet, ok := mappedSubnets[name]; !ok {
-			fctx.inventory.Delete(*subnet.ID)
+	for _, resource := range fctx.inventory.ByKind(KindSubnet) {
+		if _, ok := mappedSubnets[resource.Name]; !ok {
+			log.V(1).Info("removing subnet from inventory", "id", resource.String())
```

Comment:
No need to explicitly invoke `.String()`, the library should do it already.

```suggestion
			log.V(1).Info("Removing subnet from inventory", "id", resource)
```

Reply:
In the end, I decided to keep the `.String()`. The `String()` implementation here contains quite some logic to rebuild the actual ID string from the struct. and without the function call it prints the struct similar to a `%v` call.

Reply:
Ok, it makes sense to keep `.String()` in that case.